src/net/ipv4.c

```diff
@@ -448,16 +451,15 @@ int ipv4_has_any_addr ( struct net_device *netdev ) {
  * @ret has_addr	Network device has this IPv4 address
  */
 static int ipv4_has_addr ( struct net_device *netdev, struct in_addr addr ) {
-	struct ipv4_miniroute *miniroute;
 
-	list_for_each_entry ( miniroute, &ipv4_miniroutes, list ) {
-		if ( ( miniroute->netdev == netdev ) &&
-		     ( miniroute->address.s_addr == addr.s_addr ) ) {
-			/* Found matching address */
-			return 1;
-		}
-	}
-	return 0;
+    struct in_addr address;
+    struct settings *settings = netdev_settings ( netdev );
+    fetch_ipv4_setting ( settings, &ip_setting, &address );
```

Comment:
This code is on the receive datapath, which is latency sensitive and cannot therefore perform operations such as fetching settings.